content-src/lib/selectLayoutRender.js

```diff
@@ -108,9 +127,11 @@ export const selectLayoutRender = (state, prefs, rickRollCache) => {
       for (const component of row.components) {
```

Comment:
We should put the filter here
const component of row.components.filter(c => !filterArray.includes(c.type))) {

See a bug that exists in master and totally ok submitting fix as separate patch. This looks like fallout of DS Sequential load and followup fix https://github.com/mozilla/activity-stream/pull/4997/files

To replicate the issue, set hardcoded_layout to false and try hiding topsites. It will fail to filter TopSites. We are only filtering TopSites from components in multiple layouts ( e.g. hard coded layout) and not from multiple components inside a single layout ( e.g. basic variant).

Reply:
I'm not sure I'm following totally.

Also, I don't know if I did the steps to test right, as topsites filtered fine when I used a non hard coded layout.

Reply:
I see, I was trying by clicking the collapse button, but if I disable it in the config, I see the issue you see now.

Nice catch.

Reply:
If I add the change you suggested, we regress 1550053

I don't fully understand why hardcoded seems to make a difference.

It looks like with one fix, we break hardcoded false, and with the other, we only break hardcoded true.

Reply:
Ok, cool, looks like it's not an issue with either this patch, or with hardcoded or not layouts.

The issue seems to be between basic and the hardcoded layout.

On master, if you see the layout to not be hardcoded, and set it to basic, you see the issues you describe above, if you set it to not be hardcoded but set the layout to the same layout that is hardcoded, it works.

Here is the pref for non hardcoded layout that's the same as the hardcoded layout. :S
`{"api_key_pref":"extensions.pocket.oAuthConsumerKey","collapsible":true,"enabled":true,"show_spocs":false,"hardcoded_layout":false,"personalized":false,"layout_endpoint":"https://getpocket.cdn.mozilla.net/v3/newtab/layout?version=1&consumer_key=$apiKey&layout_variant=3-col-1-row-octr-4-col-6-row-categories"}`

Reply:
So something up with basic on master. Dunno if we want to regress 1550053 and if we do want that change, we probably want it in another ticket.

Reply:
Here's the difference I see in layout feeds:
For basic https://getpocket.cdn.mozilla.net/v3/newtab/layout?version=1&consumer_key=40249-e88c401e1b1f2242d9e441c4&layout_variant=basic

For hardcoded layout or a complex variant: https://getpocket.cdn.mozilla.net/v3/newtab/layout?version=1&consumer_key=40249-e88c401e1b1f2242d9e441c4&layout_variant=66-release-2-above-3-below-4
May be its mathijs questions why basic variant has single layout component with multiple types inside it?

Reply:
I see. If I have the filter in both spots, things probably work.

Reply:
https://bugzilla.mozilla.org/show_bug.cgi?id=1552237